listener/listener.go

```diff
@@ -63,89 +63,130 @@ func (l *listener) ProcessEvent(w http.ResponseWriter, request *http.Request) {
 		log.Error("error reading webhook event", zap.Error(err))
 		return
 	}
+
+	// only process events that occur when scanning is finished
+	if event.Type != harbor.SCANNING_FAILED && event.Type != harbor.SCANNING_COMPLETED {
+		w.WriteHeader(http.StatusOK)
+		log.Debug(fmt.Sprintf("not processing event with type %s", event.Type))
+		return
+	}
+
 	log = log.With(zap.Any("harbor event", event))
 	log.Debug("received event from harbor")
 
-	var occurrences []*grafeas_go_proto.Occurrence
+	for _, resource := range event.Data.Resources {
+		if resource.ScanOverview.Report == nil {
+			log.Error("expected event resource to contain a report")
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
 
-	if event.Type == harbor.PUSH_ARTIFACT || event.Type == harbor.SCANNING_FAILED {
-		occurrences = append(occurrences, createDiscoveryOccurrence(event))
-	}
-
-	if event.Type == harbor.SCANNING_COMPLETED {
-		occurrences = append(occurrences, createDiscoveryOccurrence(event))
-		report := event.Data.Resources[0].ScanOverview.Report
-		if report != nil && report.Summary.Total > 0 {
-			scanOccurrences, err := l.createVulnerabilityOccurrences(event)
-			if err != nil {
-				log.Error("error creating occurrences for vulnerabilities", zap.Error(err))
-				w.WriteHeader(http.StatusInternalServerError)
-				return
-			}
-			occurrences = append(occurrences, scanOccurrences...)
+		err := l.processEventResource(event, resource)
+		if err != nil {
+			log.Error("error processing event resource", zap.Error(err))
+			w.WriteHeader(http.StatusInternalServerError)
 		}
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	w.WriteHeader(http.StatusOK)
+}
+
+func (l *listener) processEventResource(event *harbor.Event, resource *harbor.Resource) error {
+	log := l.logger.Named("processEventResource").With(zap.Any("resource", resource))
+	log.Debug("processing event resource")
+
+	report := resource.ScanOverview.Report
+
+	// allow for one minute to process this event resource
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
```

Comment:
Does Harbor expect webhooks to return in a certain amount of time?

Reply:
Just tested it, harbor doesn't seem to care if the response takes longer than a minute.